go-controller/pkg/ovn/namespace.go

```diff
@@ -409,8 +410,38 @@ func (oc *Controller) deleteNamespaceLocked(ns string) *namespaceInfo {
 }
 
 func (oc *Controller) createNamespaceAddrSetAllPods(ns string) (addressset.AddressSet, error) {
-	// Get all the pods in the namespace and append their IP to the address_set
 	var ips []net.IP
+	// special handling of host network namespace
+	if config.Kubernetes.HostNetworkNamespace != "" &&
+		ns == config.Kubernetes.HostNetworkNamespace {
+		// add the mp0 interface addresses to this namespace.
+		existingNodes, err := oc.watchFactory.GetNodes()
+		if err != nil {
+			klog.Errorf("Failed to get all nodes (%v)", err)
+		} else {
+			ips = make([]net.IP, 0, len(existingNodes))
```

Comment:
curious why did u set capacity to # of nodes while we know for every node we could have multiple IPs so the slice will get resized multiple times ?

Reply:
Often times, most clusters are single-stack deployments and that is a good heuristic to start. The node.Status.Addresses field can have arbitrary number of node IPs based on  which NICS are exposed as IP addresses. Even if there is a chance that the slice will get resized, it will not happen multiple times. The runtime will hopefully increase the capacity not just by 1 every time it falls short.